src/bp/core/services/telemetry/sdk-methods.ts

```diff
@@ -0,0 +1,188 @@
+    const hooksPayload = await getHooksLifecycle(this.botService, this.ghostService)
+    const global = await this.parseHooks(hooksPayload.global.filter(hook => hook.type === 'custom'))
+    const perBots = await Promise.reduce(hooksPayload.perBots, this.botHooksReducer(), [] as ParsedFile[])
+
+    return { global, perBots }
+  }
+
+  private botHooksReducer(): (parsedFilesAcc: ParsedFile[], botHooks: BotHooks) => Promise<ParsedFile[]> {
+    return async (parsedFilesAcc, botHooks: BotHooks) => {
+      const { botId, hooks } = botHooks
+      const parsedFiles = await this.parseHooks(
+        hooks.filter(hook => hook.type === 'custom'),
+        botId
+      )
+      return [...parsedFilesAcc, ...parsedFiles]
+    }
+  }
+
+  private async parseHooks(hooks: Hook[], botId?: string): Promise<ParsedFile[]> {
+    return Promise.map(hooks, async hook => {
+      const { lifecycle, enabled, path } = hook
+      const usageParams = { lifecycle, enabled }
+
+      if (botId) {
+        _.set(usageParams, 'botId', botId)
+      }
+
+      return this.parseFile('/hooks', usageParams)(path!)
+    })
+  }
+
+  private parseFile(rootFolder: string, usageParams?: UsageParams): (name: string) => Promise<ParsedFile> {
+    return async (name: string) => {
+      const file = await this.readFileAsString(rootFolder, name, _.get(usageParams, 'botId'))
+      const functions = this.extractFunctions(parse(file, PARSE_CONFIG))
+      const usage: ParsedFile = { fileName: name.split('/').pop() || '', usages: this.parseFunctions(functions) }
+      return { ...usage, ...usageParams }
+    }
+  }
+
+  private parseFunctions(functions: string[]): Usage[] {
+    const sdkFunctions = _.countBy(functions.filter(method => method.split('.')[0] === 'bp'))
+    return _.map(sdkFunctions, (count, fn) => {
+      const [, ...namespace] = fn.split('.')
+      const functionName = namespace.pop() || ''
+      return { namespace: namespace.join('.') || '', function: functionName, count }
+    })
+  }
+
+  private extractFunctions(ast): string[] {
+    const functions: string[] = []
+
+    const findCallee = function(node) {
+      if (node.type === 'MemberExpression') {
+        return `${findCallee(node.object)}.${node.property.name}`
+      } else {
+        return node.name // breaks recursion
+      }
+    }
+
+    const enter = function(path) {
+      if (path.node.type === 'CallExpression') {
+        const callee = findCallee(path.node.callee)
+        functions.push(callee)
+      }
+    }
+
+    traverse(ast, { enter })
+
+    return functions
+  }
+
+  private async readFileAsString(rootFolder: string, fileName: string, botId?: string): Promise<string> {
+    try {
+      if (botId) {
+        return await this.ghostService.forBot(botId).readFileAsString(rootFolder, fileName)
+      } else {
+        return await this.ghostService.global().readFileAsString(rootFolder, fileName)
+      }
+    } catch (error) {
+      return ''
+    }
+  }
```

Comment:
Don't await when returning.
Also, cleaner: 
```js
const ghost = botId ? this.ghostService.forBot(botId) : this.ghostService.global()
return ghost.readFileAsString(rootFolder, fileName)
```

Reply:
I need to await if I want the error the be caught by the try/catch